src/hotspot/share/opto/memnode.cpp

```diff
@@ -1962,7 +2511,21 @@ Node *LoadNode::Ideal(PhaseGVN *phase, bool can_reshape) {
     }
   }
 
-  return progress ? this : nullptr;
+  if (progress) {
+    return this;
+  }
+
+  if (MergeLoads && UseUnalignedAccesses) {
+    if (phase->C->merge_memops_phase()) {
+      MergePrimitiveLoads merge(phase, this);
+      Node* merged = merge.run();
+      if (merged != nullptr) { return merged; }
```

Comment:
I'm a little confused here... So imagine we have a `LoadB` here. How can we now return a `LoadI` instead, and replace all uses of the `LoadB` with the `LoadI`? Should we not be replacing the `OrI` instead?

Reply:
Can you please elaborate? I'm not understanding what you are saying.

Reply:
For `LoadB` nodes, I will check they have unique usage to `OrI`. If they can be merged, the merged `LoadI` will replace the last `OrI` node, and all `LoadB` and `OrI` nodes will be dead code.
I'm not sure if I can return the merged `LoadI` here.